src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -75,16 +75,24 @@ public void RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment rend
             batchBuilder.UpdatedComponentDiffs.Append(diff);
         }
 
-        public void DisposeInBatch(RenderBatchBuilder batchBuilder)
+        public bool TryDisposeInBatch(RenderBatchBuilder batchBuilder, out Exception exception)
         {
             _componentWasDisposed = true;
+            exception = null;
 
-            // TODO: Handle components throwing during dispose. Shouldn't break the whole render batch.
-            if (Component is IDisposable disposable)
+            try
```

Comment:
I think the reason we didn't `try`/`catch` this before was that we expected that, if any of the `Dispose` calls throws, then that's going to bubble up to be an unhandled exception on the renderer, which is fatal and will dispose everything (including all components) anyway.

Was that reasoning wrong? Could you clarify why?

Reply:
This is needed because we now the requirement that we attempt to dispose *all* components rather than halting if one fails. 

If it makes the code clearer, I can move the try/catch into the caller.

Reply:
> This is needed because we now the requirement that we attempt to dispose all components rather than halting if one fails.

Sorry if I was unclear. I'm asking whether we already get that behavior automatically since, if any single component `Dispose` throws, that will be a fatal unhandled exception that will terminate the circuit, dispose the renderer, and in turn dispose all its other components anyway.

Reply:
I'm not sure that's right, I'll add a test and investigate. 

I expect exactly one of the following is true:
- Throwing during dispose will dispose some components twice
- Throwing during dispose will dispose some components 0 times
- The renderer tracks which components threw and didn't so that it can ensure they are each disposed once

Since we know that the option 3 isn't true in the checked in code, then either one of the others is true, and neither is ideal.

Reply:
So what happens if I don't try/catch around this is that it breaks the render batch, as-in one one batch is present after rendering twice.

I've added a test in a new commit that verifies every component is disposed exactly once regardless of throwing, or order relative to things throwing.

If you feel like we need to discuss this further, let's hop on a call and you can tell me what you have in mind as an alternative.

Reply:
That makes sense. Thanks for clarifying this.